src/argilla/client/feedback/metrics/agreement_metrics.py

```diff
@@ -0,0 +1,273 @@
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.client.feedback.schemas.enums import ResponseStatusFilter
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+
+
+# Expected format for the nltk's AnnotationTask
+FormattedResponses = List[Tuple[Any, Hashable, Hashable]]
+
+
+class AgreementMetricResult(BaseModel):
+    """Container for the result of an agreement metric.
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: float
+
+
+def prepare_dataset_for_annotation_task(
+    dataset: Union["FeedbackDataset", "RemoteFeedbackDataset"], question_name: str
+) -> FormattedResponses:
+    """Helper function to prepare the dataset for the nltk's AnnotationTask.
+
+    The AnnotationTask class from nltk expects the data to be formatted as a list
+    of tuples, each containing the annotator id, the task id and the label.
+
+    The AnnotationTask is supposed to deal with sets and hashable objects, but
+    there are errors transforming the data to sets and using the MASI distance function.
+    For the moment what we do with those type of questions is create a string
+    with all the values, and use the edit distance function.
+
+    Note:
+        We could potentially extend the functionality to a more than a question name. The
+        requirement would be that all the questions are of the same type, as that would
+        determine the type of distance function to use.
+
+    Args:
+        dataset: FeedbackDataset to compute the metrics.
+        question_name: Name of the question for which we want to analyse the agreement.
+
+    Returns:
+        formatted_responses: The responses formatted as a list of tuples of (user_id, question_id, value).
+    """
+    question_type = type(dataset.question_by_name(question_name))
+    supported_question_types = list(QUESTION_TO_DISTANCE.keys())
+    if question_type not in supported_question_types:
+        raise NotImplementedError(
+            f"Question '{question_name}' is of type '{question_type}', the supported question types are: {supported_question_types}."
+        )
+
+    dataset = dataset.filter_by(response_status=ResponseStatusFilter.submitted.value)
+
+    hf_dataset = dataset.format_as("datasets")
+
+    formatted_responses: FormattedResponses = []
+
+    for responses_ in hf_dataset[question_name]:
+        for response in responses_:
+            # We do this check here because local datasets don't implement the filter_by method.
+            if response["status"] != ResponseStatusFilter.submitted.value:
+                continue
+            user_id = response["user_id"]
+
+            if user_id is None:
+                raise NotImplementedError(
+                    "In order to use this functionality the records need to be assigned to a user."
```

Comment:
I believe this is done automatically when pushed to Argilla, so we might encourage them to push it and continue after instead of this less explicit error?